lib/Dialect/CGGI/Transforms/BooleanLineVectorizer.cpp

```diff
@@ -30,15 +42,126 @@ namespace cggi {
   // on the CGGI_BinaryGateOp's?
   return OpTrait::hasElementwiseMappableTraits(lhs);
 }
 
-bool tryBoolVectorizeBlock(Block *block, MLIRContext &context) {
+FailureOr<SmallVector<Attribute>> BuildGateOperands(
+    const SmallVector<Operation *> &bucket, MLIRContext &context) {
+  SmallVector<Attribute> vectorizedGateOperands;
+  for (auto *op : bucket) {
+    FailureOr<Attribute> attr =
+        llvm::TypeSwitch<Operation &, FailureOr<Attribute>>(*op)
+            .Case<cggi::AndOp>(
+                [&](auto op) { return StringAttr::get(&context, "and"); })
+            .Case<cggi::NandOp>(
+                [&](auto op) { return StringAttr::get(&context, "nand"); })
+            .Case<cggi::XorOp>(
+                [&](auto op) { return StringAttr::get(&context, "xor"); })
+            .Case<cggi::XNorOp>(
+                [&](auto op) { return StringAttr::get(&context, "xnor"); })
+            .Case<cggi::Lut3Op>(
+                [&](cggi::Lut3Op op) { return op.getLookupTable(); })
+            .Default([&](Operation &op) -> FailureOr<Attribute> {
+              op.emitOpError("Unable to find printer for op");
+              return failure();
+            });
+    if (failed(attr)) {
+      op->emitOpError("Failed attr");
+      return failure();
+    }
+    vectorizedGateOperands.push_back(attr.value());
+  }
+  LLVM_DEBUG({
+    llvm::dbgs() << "Go over vectorizedGateOps:\n";
+    for (auto op : vectorizedGateOperands) {
+      llvm::dbgs() << " - " << op;
+    }
+    llvm::dbgs() << "\n";
+  });
+  return vectorizedGateOperands;
+}
+
+SmallVector<Value> BuildVectorizedOperands(
+    Operation *key, const SmallVector<Operation *> &bucket,
+    RankedTensorType tensorType, OpBuilder builder) {
+  SmallVector<Value> vectorizedOperands;
+  // Group the independent operands over the operations
+  for (uint operandIndex = 0; operandIndex < key->getNumOperands();
+       ++operandIndex) {
+    SmallVector<Value, 4> operands;
+    LLVM_DEBUG({
+      llvm::dbgs() << "For: " << key->getName()
+                   << " Number of ops: " << key->getNumOperands() << "\n";
+    });
+
+    operands.reserve(bucket.size());
+    ///------------------------------------------
+    for (auto *op : bucket) {
+      LLVM_DEBUG(llvm::dbgs() << "getOperand for [" << operandIndex
+                              << "]: " << op->getOperand(operandIndex) << "\n");
+      operands.push_back(op->getOperand(operandIndex));
+    }
+    ///------------------------------------------
+    auto fromElementsOp = builder.create<tensor::FromElementsOp>(
+        key->getLoc(), tensorType, operands);
+    vectorizedOperands.push_back(fromElementsOp.getResult());
+  }
+  LLVM_DEBUG({
+    llvm::dbgs() << "Go over vectorizedOps:\n";
+    for (auto op : vectorizedOperands) {
+      llvm::dbgs() << " - " << op << "\n";
+    }
+  });
+  return vectorizedOperands;
+}
+
+int bucket_size(const SmallVector<SmallVector<Operation *>> &buckets) {
```

Comment:
nit: in this file there are function patterns like "BuildGate", "buildCompatible", and "bucket_size" - could you keep them all as camelcase?

Reply:
Done!